packages/components/nodes/vectorstores/InMemory/InMemoryVectorStore.ts

```diff
@@ -71,7 +71,9 @@ class InMemoryVectorStore_VectorStores implements INode {
         const flattenDocs = docs && docs.length ? flatten(docs) : []
         const finalDocs = []
         for (let i = 0; i < flattenDocs.length; i += 1) {
-            finalDocs.push(new Document(flattenDocs[i]))
+            if (flattenDocs[i] !== undefined) {
+                finalDocs.push(new Document(flattenDocs[i]))
```

Comment:
yeah you are right. The error occurs when you have undefined `pageContent`. However this won't fix it because `flattenDocs[i]` is an object:
```
{
            pageContent: 'example',
            metadata: {
                url: 'pageUrl'
            }
        }
```

I think the correct fix is to check if the `pageContent` is undefined. To make it extra secure we can check `if (flattenDocs[i] !== undefined && flattenDocs[i].pageContent !== undefined)`